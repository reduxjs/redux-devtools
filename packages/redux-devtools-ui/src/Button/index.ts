export { default } from './Button.js';
