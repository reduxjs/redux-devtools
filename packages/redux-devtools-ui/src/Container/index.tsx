import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { useTheme, ThemeData } from '../utils/theme';
import { MainContainerWrapper, ContainerWrapper } from './styles';
import { Theme } from '../themes/default';

interface ContainerFromThemeDataProps {
  children?: React.ReactNode;
  themeData: ThemeData;
  className?: string;
}

const ContainerFromThemeData: React.FunctionComponent<
  ContainerFromThemeDataProps
> = ({ themeData, className, children }) => {
  const theme = useTheme(themeData);
  return (
    <ThemeProvider theme={theme}>
      <MainContainerWrapper className={className}>
        {children}
      </MainContainerWrapper>
    </ThemeProvider>
  );
};

interface Props {
  children?: React.ReactNode;
  themeData?: ThemeData;
  theme?: Theme;
  className?: string;
}

const Container: React.FunctionComponent<Props> = ({
  themeData,
  className,
  theme,
  children,
}) => {
  if (!themeData) {
    return (
      <ContainerWrapper className={className} theme={theme}>
        {children}
      </ContainerWrapper>
    );
  }

  return (
    <ContainerFromThemeData themeData={themeData} className={className}>
      {children}
    </ContainerFromThemeData>
  );
};

Container.propTypes = {
  children: PropTypes.node,
  themeData: PropTypes.any,
  theme: PropTypes.any,
  className: PropTypes.string,
};

export default Container;
