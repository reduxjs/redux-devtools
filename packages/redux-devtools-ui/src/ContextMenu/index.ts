export { default } from './ContextMenu.js';
