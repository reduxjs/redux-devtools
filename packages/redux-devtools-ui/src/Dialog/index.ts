export { default } from './Dialog.js';
