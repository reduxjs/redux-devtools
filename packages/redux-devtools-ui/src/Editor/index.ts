export { default } from './Editor.js';
