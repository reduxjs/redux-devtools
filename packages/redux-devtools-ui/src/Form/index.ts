export { default } from './Form.js';
