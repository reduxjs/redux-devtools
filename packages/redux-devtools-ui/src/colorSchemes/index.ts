export { default } from './default';
export { default as github } from './github';
export { default as atomOneDark } from './atom-one-dark';
export { default as dracula } from './dracula';
export { default as iRBlack } from './ir-black';
export { default as macintosh } from './macintosh';
export { default as materia } from './materia';
export { default as oceanicNext } from './oceanic-next';
export { default as phD } from './phd';
export { default as pico } from './pico';
export { default as solarFlare } from './solar-flare';
export { default as spacemacs } from './spacemacs';
export { default as unikitty } from './unikitty';
export { default as woodland } from './woodland';
