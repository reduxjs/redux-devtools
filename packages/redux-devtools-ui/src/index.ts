export { default as Container } from './Container';
export { default as Button } from './Button';
export { default as ContextMenu } from './ContextMenu';
export { default as Dialog } from './Dialog';
export { default as Editor } from './Editor';
export { default as Form } from './Form';
export { default as Select } from './Select';
export { default as Slider } from './Slider';
export { default as Tabs, Tab } from './Tabs';
export { default as SegmentedControl } from './SegmentedControl';
export { default as Notification } from './Notification';
export * from './Toolbar';

import color from './utils/color';
export const effects = { color };
export { default as createStyledComponent } from './utils/createStyledComponent';
export { Theme, ThemeFromProvider, Scheme } from './utils/theme';
