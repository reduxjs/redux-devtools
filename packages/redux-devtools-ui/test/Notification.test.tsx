import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Notification } from '../src';

describe('Notification', function () {
  it('renders correctly', () => {
    const { container } = render(<Notification>Message</Notification>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with props', () => {
    const { container } = render(
      <Notification
        type="error"
        onClose={() => {
          // noop
        }}
      >
        Message
      </Notification>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should handle the click event', () => {
    const onClose = jest.fn();
    render(<Notification onClose={onClose}>Message</Notification>);

    userEvent.click(screen.getByRole('button'));
    expect(onClose).toHaveBeenCalled();
  });
});
