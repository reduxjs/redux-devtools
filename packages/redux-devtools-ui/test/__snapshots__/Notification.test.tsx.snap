// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders correctly 1`] = `
<div
  class="sc-eDWCr fTovFd"
  type="info"
>
  <span>
    Message
  </span>
</div>
`;

exports[`Notification renders with props 1`] = `
<div
  class="sc-eDWCr fTovFd"
  type="error"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
    />
  </svg>
  <span>
    Message
  </span>
  <button>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
  </button>
</div>
`;
